filebeat/input/systemlogs/input.go

```diff
@@ -0,0 +1,234 @@
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package systemlogs
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"github.com/elastic/beats/v7/filebeat/channel"
+	v1 "github.com/elastic/beats/v7/filebeat/input"
+	loginput "github.com/elastic/beats/v7/filebeat/input/log"
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	cursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/feature"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+const pluginName = "system-logs"
+
+func init() {
+	// Register an input V1, that's used by the log input
+	if err := v1.Register(pluginName, newV1Input); err != nil {
+		panic(err)
+	}
+}
+
+type config struct {
+	UseJournald bool    `config:"use_journald"`
+	UseFiles    bool    `config:"use_files"`
+	Files       *conf.C `config:"files" yaml:"files"`
+	Journald    *conf.C `config:"journald" yaml:"journald"`
+}
+
+func (c *config) Validate() error {
+	if c.UseFiles && c.UseJournald {
+		return errors.New("'use_journald' and 'use_files' cannot both be true")
+	}
+
+	if c.Files == nil && c.Journald == nil {
+		return errors.New("one of 'journald' or 'files' must be set")
+	}
+
+	return nil
+}
+
+// newV1Input checks whether the log input must be created and
+// delegates to loginput.NewInput if needed.
+func newV1Input(
+	cfg *conf.C,
+	outlet channel.Connector,
+	context v1.Context,
+) (v1.Input, error) {
+	journald, err := useJournald(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("cannot decide between journald and files: %w", err)
+	}
+
+	if journald {
+		return nil, v2.ErrUnknownInput
+	}
+
+	// Convert the configuration and create a log input
+	logCfg, err := toFilesConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return loginput.NewInput(logCfg, outlet, context)
+}
+
+// PluginV2 creates a v2.Plugin that will instantiate a journald
+// input if needed.
+func PluginV2(logger *logp.Logger, store cursor.StateStore) v2.Plugin {
+	logger = logger.Named(pluginName)
+
+	return v2.Plugin{
+		Name:       pluginName,
+		Stability:  feature.Stable,
+		Deprecated: false,
+		Info:       "system-logs input",
+		Doc:        "The system-logs input collects system logs on Linux by reading them from journald or traditional log files",
+		Manager: &cursor.InputManager{
+			Logger:     logger,
+			StateStore: store,
+			Type:       pluginName,
+			Configure:  configure,
+		},
+	}
+}
+
+// useJournald returns true if jounrald should be used.
+// If there is an error, false is always retruned.
+//
+// The decision logic is:
+//   - If UseJournald is set, return true
+//   - If UseFiles is set, return false
+//   - If the globs defined in `files.paths` match any existing file,
+//     return false
+//   - Otherwise return true
+func useJournald(c *conf.C) (bool, error) {
```

Comment:
This function would benefit from some logging to indicate why it made the choice it did.

Reply:
Indeed it does, thanks!

Done: https://github.com/elastic/beats/pull/41246